src/n_fft/idft.c

```diff
@@ -0,0 +1,321 @@
+                                                            \
+    IDFT4_NODE0_LAZY14(p0, p1, p2, p3,                      \
+                       tab_w[2], tab_w[3],                  \
+                       mod, mod2, p_hi, p_lo);              \
+    IDFT4_LAZY12(p4, p5, p6, p7,                            \
+                tab_w[2], tab_w[3],                         \
+                tab_w[4], tab_w[5],                         \
+                tab_w[6], tab_w[7],                         \
+                mod, mod2, p_hi, p_lo);                     \
+                                                            \
+    BUTTERFLY_LAZY4244(p0, p4, mod2, tmp);                  \
+    BUTTERFLY_LAZY4244(p1, p5, mod2, tmp);                  \
+    BUTTERFLY_LAZY4244(p2, p6, mod2, tmp);                  \
+    BUTTERFLY_LAZY4244(p3, p7, mod2, tmp);                  \
+} while(0)
+
+
+/** 8-point IDFT
+ * TODO clean, check laziness
+ * * in [0..?n) / out [0..?n) / max < ?n
+ */
+#define DFT8_LAZY12(p0, p1, p2, p3, p4, p5, p6, p7,              \
+                    node, mod, mod2, tab_w)                      \
+do {                                                             \
+    ulong p_hi, p_lo, tmp;                                       \
+                                                                 \
+    const ulong w = tab_w[2*(node)];                             \
+    const ulong w_pr = tab_w[2*(node)+1];                        \
+                                                                 \
+    IDFT4_LAZY12(p0, p1, p2, p3,                                 \
+                 tab_w[4*(node)], tab_w[4*(node)+1],             \
+                 tab_w[8*(node)], tab_w[8*(node)+1],             \
+                 tab_w[8*(node)+2], tab_w[8*(node)+3],           \
+                 mod, mod2, p_hi, p_lo);                         \
+                                                                 \
+    IDFT4_LAZY12(p4, p5, p6, p7,                                 \
+                 tab_w[4*(node)+2], tab_w[4*(node)+3],           \
+                 tab_w[8*(node)+4], tab_w[8*(node)+5],           \
+                 tab_w[8*(node)+6], tab_w[8*(node)+7],           \
+                 mod, mod2, p_hi, p_lo);                         \
+                                                                 \
+    IDFT2_LAZY22(p0, p4, mod, mod2, w, w_pr, p_hi, p_lo, tmp);  \
+    IDFT2_LAZY22(p1, p5, mod, mod2, w, w_pr, p_hi, p_lo, tmp);  \
+    IDFT2_LAZY22(p2, p6, mod, mod2, w, w_pr, p_hi, p_lo, tmp);  \
+    IDFT2_LAZY22(p3, p7, mod, mod2, w, w_pr, p_hi, p_lo, tmp);  \
+} while(0)
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*--------------*/
+/* general IDFT */
+/*--------------*/
+
+
+// TODO doc
+void idft_lazy12(nn_ptr p, ulong depth, ulong node, n_fft_args_t F)
+{
+    if (depth == 1)
+    {
+        ulong p_hi, p_lo, tmp;
+        IDFT2_LAZY22(p[0], p[1], F->mod, F->mod2, F->tab_w[2*node], F->tab_w[2*node+1], p_hi, p_lo, tmp);
+    }
```

Comment:
Codecov complains about not reaching these lines. Is it possible to reach these?

Reply:
Yes, these lines are not reached right now because only the `node0` variants are tested currently (they do call the function containing these lines but never with `depth == 1`). So these lines will be reached once the tests are more complete. I'm adding a todo in the code to make I do not forget about this. Thanks for the catch!